x-pack/plugins/security_solution/public/cases/components/create/form_context.test.tsx

```diff
@@ -0,0 +1,470 @@
+
+const defaultPostCase = {
+  isLoading: false,
+  isError: false,
+  caseData: null,
+  postCase,
+};
+
+const fillForm = async (wrapper: ReactWrapper) => {
+  await act(async () => {
+    wrapper
+      .find(`[data-test-subj="caseTitle"] input`)
+      .first()
+      .simulate('change', { target: { value: sampleData.title } });
+  });
+
+  await act(async () => {
+    wrapper
+      .find(`[data-test-subj="caseDescription"] textarea`)
+      .first()
+      .simulate('change', { target: { value: sampleData.description } });
+  });
+
+  await waitFor(() => {
+    ((wrapper.find(EuiComboBox).props() as unknown) as {
+      onChange: (a: EuiComboBoxOptionOption[]) => void;
+    }).onChange(sampleTags.map((tag) => ({ label: tag })));
+  });
+};
+
+describe('Create case', () => {
+  const fetchTags = jest.fn();
+  const onFormSubmitSuccess = jest.fn();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    usePostCaseMock.mockImplementation(() => defaultPostCase);
+    useConnectorsMock.mockReturnValue(sampleConnectorData);
+    useCaseConfigureMock.mockImplementation(() => useCaseConfigureResponse);
+    useGetIncidentTypesMock.mockReturnValue(useGetIncidentTypesResponse);
+    useGetSeverityMock.mockReturnValue(useGetSeverityResponse);
+    useGetIssueTypesMock.mockReturnValue(useGetIssueTypesResponse);
+    useGetFieldsByIssueTypeMock.mockReturnValue(useGetFieldsByIssueTypeResponse);
+
+    (useGetTags as jest.Mock).mockImplementation(() => ({
+      tags: sampleTags,
+      fetchTags,
+    }));
+  });
+
+  describe('Step 1 - Case Fields', () => {
+    it('it renders', async () => {
+      const wrapper = mount(
+        <TestProviders>
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseForm />
+            <SubmitCaseButton />
+          </FormContext>
+        </TestProviders>
+      );
+
+      expect(wrapper.find(`[data-test-subj="caseTitle"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseDescription"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseTags"]`).first().exists()).toBeTruthy();
+      expect(wrapper.find(`[data-test-subj="caseConnectors"]`).first().exists()).toBeTruthy();
+      expect(
+        wrapper.find(`[data-test-subj="case-creation-form-steps"]`).first().exists()
+      ).toBeTruthy();
+    });
+
+    it('should post case on submit click', async () => {
+      useConnectorsMock.mockReturnValue({
+        ...sampleConnectorData,
+        connectors: connectorsMock,
+      });
+
+      const wrapper = mount(
+        <TestProviders>
+          <FormContext onSuccess={onFormSubmitSuccess}>
+            <CreateCaseForm />
+            <SubmitCaseButton />
+          </FormContext>
+        </TestProviders>
+      );
+
+      await fillForm(wrapper);
+      wrapper.update();
+
+      await act(async () => {
+        wrapper.find(`[data-test-subj="create-case-submit"]`).first().simulate('click');
+      });
```

Comment:
do not need update or act here
```suggestion
      wrapper.find(`[data-test-subj="create-case-submit"]`).first().simulate('click');
```

Reply:
please do this for all instances of this pattern in this file

Reply:
For one test the act is needed to be passed. We need to `await` for all promises to be fulfilled.